<a href="https://colab.research.google.com/github/argg1006/TRABAJOS/blob/main/02_ebusiness.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Taller: Auditoría de Desempeño para Negocios Digitales:

## Parte 1: Análisis de datos sintéticos

En esta parte vamos a analizar un dataset de transacciones y determinar si la estrategia de M-Commerce (App Móvil) es más rentable que el E-Commerce tradicional (Web Desktop).

Vamos a generar una base de datos aleatoria, que simula la salida de un TPS.

In [ ]:
import pandas as pd
import numpy as np

# Generar datos simulados
data = {
    'ID_Transaccion': range(1001, 1051),
    'Canal': np.random.choice(['Web Desktop', 'App Móvil'], 50, p=[0.4, 0.6]),
    'Metodo_Pago': np.random.choice(['Tarjeta Crédito', 'PayPal', 'Transferencia'], 50),
    'Total_Venta': np.random.randint(20, 500, 50), # Ventas entre $20 y $500
    'Categoria': np.random.choice(['Electrónica', 'Moda', 'Hogar'], 50)
}

df = pd.DataFrame(data)

df


,ID_Transaccion,Canal,Metodo_Pago,Total_Venta,Categoria
0,1001,Web Desktop,Tarjeta Crédito,252,Electrónica
1,1002,App Móvil,PayPal,90,Hogar
2,1003,App Móvil,Tarjeta Crédito,394,Moda
3,1004,Web Desktop,Transferencia,444,Electrónica
4,1005,App Móvil,Tarjeta Crédito,295,Electrónica
5,1006,Web Desktop,Tarjeta Crédito,453,Moda
6,1007,App Móvil,Transferencia,274,Hogar
7,1008,App Móvil,Transferencia,198,Moda
8,1009,Web Desktop,PayPal,358,Electrónica
9,1010,App Móvil,Tarjeta Crédito,419,Moda


### ¿Qué preguntas deberíamos plantearnos para obtener las respuestas que buscamos?

Buscamos saber cuál de las dos estrategias (M-Commerce o E-Commerce) es la más rentable.

Vamos a crear 4 KPIs para un MIS que nos permitan responder esta pregunta.

In [ ]:
# KPI 1: Ingreso Total por Canal
kpi1 = df.groupby('Canal')['Total_Venta'].sum()
print("Ingreso Total por Canal ")
print(kpi1)
print()

# KPI 2: Ticket Promedio por Canal
kpi2 = df.groupby('Canal')['Total_Venta'].mean().round(2)
print("Ticket Promedio por Canal")
print(kpi2)
print()

# KPI 3: Número de Transacciones por Canal
kpi3 = df.groupby('Canal')['ID_Transaccion'].count()
print("=== KPI 3: Número de Transacciones por Canal ===")
print(kpi3)
print()

# KPI 4: Ingreso por Categoría y Canal
kpi4 = df.groupby(['Canal', 'Categoria'])['Total_Venta'].sum()

print("Ingreso por Categoría y Canal")
print(kpi4)

Ingreso Total por Canal 
Canal
App Móvil      8470
Web Desktop    5334
Name: Total_Venta, dtype: int64

Ticket Promedio por Canal
Canal
App Móvil      264.69
Web Desktop    296.33
Name: Total_Venta, dtype: float64

=== KPI 3: Número de Transacciones por Canal ===
Canal
App Móvil      32
Web Desktop    18
Name: ID_Transaccion, dtype: int64

Ingreso por Categoría y Canal
Canal        Categoria  
App Móvil    Electrónica    1561
             Hogar          2859
             Moda           4050
Web Desktop  Electrónica    2778
             Hogar           952
             Moda           1604
Name: Total_Venta, dtype: int64


**Conclusiones**


*   En base a los resultados dados, el más rentable es M-Commerce que el E-Commerce, por su comparación con los KPIS, se refleja que tiene mas ingresos por canal, a su vez el número de transacciones, e igual puede haber leves difencias en ingreso por categoría, pero sigue este teniendo más ingresos.




## Parte 2: KPIs para E-Commerce

Para esta parte vamos a utilizar los datos qu se encuentran en el repositorio en `/data/ecommerce.zip`

In [ ]:
import pandas as pd

In [ ]:
data = pd.read_csv('../data/ecommerce.zip')
data

,ID,Warehouse_block,Mode_of_Shipment,Customer_care_calls,Customer_rating,Cost_of_the_Product,Prior_purchases,Product_importance,Gender,Discount_offered,Weight_in_gms,Reached.on.Time_Y.N
0,1,D,Flight,4,2,177,3,low,F,44,1233,1
1,2,F,Flight,4,5,216,2,low,M,59,3088,1
2,3,A,Flight,2,2,183,4,low,M,48,3374,1
3,4,B,Flight,3,3,176,4,medium,M,10,1177,1
4,5,C,Flight,2,2,184,3,medium,F,46,2484,1
...,...,...,...,...,...,...,...,...,...,...,...,...
10994,10995,A,Ship,4,1,252,5,medium,F,1,1538,1
10995,10996,B,Ship,4,1,232,5,medium,F,6,1247,0
10996,10997,C,Ship,5,4,242,5,low,F,4,1155,0
10997,10998,F,Ship,5,2,223,6,medium,M,2,1210,0


### 1. KPIs de Gestión de Logística y Operaciones (Shipment & Warehouse)

Permiten medir la eficiencia en la distribución de los productos electrónicos.

1.1. **Costo Promedio de Envío por Modo de Transporte (Mode of Shipment):** Ayuda a evaluar qué canales (Barco, Avión, Carretera) representan un mayor impacto financiero en relación con el peso del producto.

1.2. **Volumen de Ocupación por Bloque de Bodega (Warehouse Block):** Porcentaje o cantidad total de productos/peso administrados en cada bloque (A, B, C, D, E) para entender la carga operativa de la infraestructura.

1.3. **Peso Promedio del Producto por Modo de Envío:** Para analizar si se está optimizando la logística (ej. si los productos más pesados se están enviando por barco o si se está gastando de más enviándolos por avión).

In [52]:
import pandas as pd
data = pd.read_csv('Train.csv')

#1.1 Costo Promedio de Envio por Modo de Transporte
Promedio=data.groupby('Mode_of_Shipment')['Cost_of_the_Product'].mean().round(2)
print("Costo Promedio de Envio por Modo de Transporte ")
print(Promedio)

#1.2 Volumen de Ocupación por Bloque de Bodega
Volumen=data.groupby('Warehouse_block')['ID'].count()
print("Volumen de Ocupación por Bloque de Bodega")
print(Volumen)

#1.3 Peso Promedio del Producto por Modo de Envío
Peso=data.groupby('Mode_of_Shipment')['Weight_in_gms'].mean().round(2)
print("Peso Promedio del Producto por Modo de Envío")
print(Peso)

Costo Promedio de Envio por Modo de Transporte 
Mode_of_Shipment
Flight    209.31
Road      210.48
Ship      210.34
Name: Cost_of_the_Product, dtype: float64
Volumen de Ocupación por Bloque de Bodega
Warehouse_block
A    1833
B    1833
C    1833
D    1834
F    3666
Name: ID, dtype: int64
Peso Promedio del Producto por Modo de Envío
Mode_of_Shipment
Flight    3629.38
Road      3649.55
Ship      3631.46
Name: Weight_in_gms, dtype: float64


**CONCLUSIONES:**

1.1 -Los tres modos tienen costos casi idénticos lo que sugiere que el precio del producto no influye en la elección del medio de envío.

1.2 -El bloque F tiene el doble de productos que todos los demás indicando que F está sobrecargado y podría ser un cuello de botella.

1.3 -El peso es casi igual en los tres modos, esto es una ineficiencia logística, lo ideal sería que el barco concentre los productos más pesados, por su capacidad.

### 2. KPIs de Gestión Comercial y Financiera (Sales & Revenue)

Enfocados en el rendimiento del dinero, los descuentos y la estrategia de precios.

2.1. **Ingreso Total Bruto y Costo Promedio del Producto:** Suma y promedio de la variable `Cost of the product` para entender el ticket promedio de la tienda de electrónica.

2.2. **Porcentaje de Descuento Promedio Otorgado:** Evaluar el impacto de la estrategia de promociones (`Discount offered`) y si los descuentos altos realmente correlacionan con clientes de mayor prioridad o compras previas.

2.3. **Margen Teórico Relativo por Importancia del Producto:** Cruzar `Product importance` (Low, Medium, High) con el costo para ver si los productos catalogados como "alta importancia" son también los más costosos o los que reciben más descuentos.

In [53]:
import pandas as pd
data = pd.read_csv('Train.csv')
#2.1 Ingreso Total Bruto y Costo Promedio
total = data['Cost_of_the_Product'].sum()
promedio = data['Cost_of_the_Product'].mean().round(2)
print("Ingreso Total Bruto y Costo Promedio")
print(f"Total bruto: ${total:,}")
print(f"Costo promedio: ${promedio}")

#2.2: Descuento Promedio según Compras Previas
Descuento = data.groupby('Prior_purchases')['Discount_offered'].mean().round(2)
print("Descuento Promedio según Compras Previas")
print(Descuento)

#KPI 2.3: Costo y Descuento por Importancia del Producto
Importancia = data.groupby('Product_importance')[['Cost_of_the_Product', 'Discount_offered']].mean().round(2)
print("Costo y Descuento por Importancia del Producto")
print(Importancia)

Ingreso Total Bruto y Costo Promedio
Total bruto: $2,311,955
Costo promedio (ticket): $210.2
Descuento Promedio según Compras Previas
Prior_purchases
2     15.32
3     14.98
4     10.85
5      9.27
6     11.06
7     17.53
8     13.32
10    13.60
Name: Discount_offered, dtype: float64
Costo y Descuento por Importancia del Producto
                    Cost_of_the_Product  Discount_offered
Product_importance                                       
high                             203.87             14.67
low                              212.06             13.04
medium                           209.38             13.48


**CONCLUSIONES:**

2.1 -Tanto el ingreso bruto total como un ticket promedio, es un número estable que sirve como línea base para medir el desempeño.

2.2 -El patrón es irregular siendo que no hay una tendencia clara, la estrategia de descuentos no está bien segmentada siendo que no premia consistentemente la lealtad.

2.3 -Los productos de alta importancia  son los más baratos y los de baja importancia son los más costosos. Eso indica que la clasificación de importancia no está alineada con el precio.

### 3. KPIs de Experiencia y Fidelización del Cliente (Customer Experience - CX)

Diseñados para evaluar la satisfacción y la carga de soporte post-venta.

3.1. **Índice de Satisfacción del Cliente (CSAT) Promedio:** El promedio de la variable `Customer rating` (escala 1 a 5). Se puede segmentar por género o por bloque de bodega para ver si hay anomalías.

3.2. **Tasa de Retención / Clientes Recurrentes:** Analizar la distribución de `Prior purchases` para determinar qué porcentaje de la base de datos son clientes nuevos (0-1 compras previas) vs. clientes fidelizados (2 o más compras).

3.3. **Carga de Soporte por Cliente (Customer Care Calls):** Promedio de llamadas de consulta realizadas. Permite analizar si a mayor costo del producto o menores calificaciones, aumentan las llamadas de reclamo.

In [54]:
import pandas as pd
data = pd.read_csv('Train.csv')
#3.1: CSAT Promedio por Género y Bloque de Bodega
Genero = data.groupby('Gender')['Customer_rating'].mean().round(2)
Bodega = data.groupby('Warehouse_block')['Customer_rating'].mean().round(2)
print("CSAT Promedio por Género")
print(Genero)
print("CSAT Promedio por Bloque de Bodega ")
print(Bodega)


#3.2: Clientes Nuevos vs Fidelizados
nuevos = (data['Prior_purchases'] <= 1).sum()
fidelizados = (data['Prior_purchases'] >= 2).sum()
total = len(data)
print("Tasa de Retención")
print(f"Clientes nuevos (0-1 compras):      {nuevos} ({round(nuevos/total*100, 1)}%)")
print(f"Clientes fidelizados (2+ compras):  {fidelizados} ({round(fidelizados/total*100, 1)}%)")

#3.3: Llamadas de Soporte Promedio por Rating
Soporte= data.groupby('Customer_rating')['Customer_care_calls'].mean().round(2)
print("Llamadas de Soporte Promedio por Rating")
print(Soporte)

CSAT Promedio por Género
Gender
F    2.99
M    2.99
Name: Customer_rating, dtype: float64
CSAT Promedio por Bloque de Bodega 
Warehouse_block
A    2.96
B    2.98
C    2.99
D    3.02
F    3.00
Name: Customer_rating, dtype: float64
Tasa de Retención
Clientes nuevos (0-1 compras):      0 (0.0%)
Clientes fidelizados (2+ compras):  10999 (100.0%)
Llamadas de Soporte Promedio por Rating
Customer_rating
1    4.04
2    4.05
3    4.06
4    4.04
5    4.09
Name: Customer_care_calls, dtype: float64



**CONCLUSIONES:**

3.1 -El rating promedio es 2.99 tanto en hombres como en mujeres, sin diferencia por género. Por bloque de bodega tampoco hay variación significativa.

3.2 -No hay clientes nuevos, solo los clientes fidelizados se registra con 100%.

3.3 -Los clientes más satisfechos  hacen más llamadas que los insatisfechos. La diferencia es mínima, lo que sugiere que el volumen de soporte no depende de la satisfacción, sino probablemente del tipo de producto o proceso de compra.